src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IArgument.cs

```diff
@@ -1359,40 +1417,244 @@ void M()
+            comp.VerifyEmitDiagnostics(expectedDiagnostics0);
+
+            var comp0 = CreateCompilation(source0);
+            comp0.VerifyEmitDiagnostics(expectedDiagnostics0);
+
+            var comp1 = CreateCompilation(source1, references: new[] { comp0.ToMetadataReference() });
+            VerifyOperationTreeAndDiagnosticsForTest<InvocationExpressionSyntax>(comp1, expectedOperationTree0And1, DiagnosticDescription.None);
+            comp1.VerifyEmitDiagnostics(DiagnosticDescription.None);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void ValidConversionForDefaultArgument_DateTime()
+        {
+            var source0 = @"
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+public class C0
+{
+    public static void M0([Optional, DateTimeConstant(634953547672667479L)] DateTime x) { } 
+}
+";
+            var source1 = @"
+public class C1
+{
+    public static void M1()
+    {
+        /*<bind>*/C0.M0()/*</bind>*/;
+    }
+}
+";
+            var expectedOperationTree0And1 = @"
+IInvocationOperation (void C0.M0([System.DateTime x])) (OperationKind.Invocation, Type: System.Void) (Syntax: 'C0.M0()')
+  Instance Receiver: 
+    null
+  Arguments(1):
+      IArgumentOperation (ArgumentKind.DefaultValue, Matching Parameter: x) (OperationKind.Argument, Type: null, IsImplicit) (Syntax: 'C0.M0()')
+        ILiteralOperation (OperationKind.Literal, Type: System.DateTime, Constant: 02/01/2013 22:32:47, IsImplicit) (Syntax: 'C0.M0()')
+        InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+        OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+";
+
+            var comp = CreateCompilation(new[] { source1, source0 });
+            VerifyOperationTreeAndDiagnosticsForTest<InvocationExpressionSyntax>(comp, expectedOperationTree0And1, DiagnosticDescription.None);
+            comp.VerifyEmitDiagnostics();
+
+            var comp0 = CreateCompilation(source0);
+            comp0.VerifyEmitDiagnostics();
+
+            var comp1 = CreateCompilation(source1, references: new[] { comp0.ToMetadataReference() });
+            VerifyOperationTreeAndDiagnosticsForTest<InvocationExpressionSyntax>(comp1, expectedOperationTree0And1, DiagnosticDescription.None);
+            comp1.VerifyEmitDiagnostics(DiagnosticDescription.None);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void InvalidConversionForDefaultArgument_DateTime()
+        {
+            var source0 = @"
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+public class C0
+{
+    public static void M0([Optional, DateTimeConstant(634953547672667479L)] string x) { } 
```

Comment:
Perhaps a quick comment as to why this literal isn't valid?

Reply:
Oh wait, duh, it's a `string` parameter.

---
In reply to: [522539595](https://github.com/dotnet/roslyn/pull/49186#discussion_r522539595) [](ancestors = 522539595)